lib/util/ts_parser.js

```diff
@@ -825,6 +835,159 @@ shaka.util.TsParser = class {
     return videoInfo;
   }
 
+  /**
+   * Return the video information for HVC
+   *
+   * @return {{height: ?string, width: ?string, codec: ?string}}
+   * @private
+   */
+  getHvcInfo_() {
+    const TsParser = shaka.util.TsParser;
+    const videoInfo = {
+      height: null,
+      width: null,
+      codec: null,
+    };
+    const videoNalus = this.getVideoNalus();
+    if (!videoNalus.length) {
+      return videoInfo;
+    }
+    const spsNalu = videoNalus.find((nalu) => {
+      return nalu.type == TsParser.H265_NALU_TYPE_SPS_;
+    });
+    if (!spsNalu) {
+      return videoInfo;
+    }
+
+    const gb = new shaka.util.ExpGolomb(
+        spsNalu.fullData, /* convertEbsp2rbsp= */ true);
+
+    // remove NALu Header
+    gb.readUnsignedByte();
+    gb.readUnsignedByte();
+
+    // SPS
+    gb.readBits(4); // video_paramter_set_id
+    const maxSubLayersMinus1 = gb.readBits(3);
+    gb.readBoolean(); // temporal_id_nesting_flag
+
+    // profile_tier_level begin
+    const generalProfileSpace = gb.readBits(2);
+    const generalTierFlag = gb.readBits(1);
+    const generalProfileIdc = gb.readBits(5);
+    const generalProfileCompatibilityFlags = gb.readBits(32);
+    const generalConstraintIndicatorFlags1 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags2 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags3 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags4 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags5 = gb.readUnsignedByte();
+    const generalConstraintIndicatorFlags6 = gb.readUnsignedByte();
+    const generalLevelIdc = gb.readUnsignedByte();
+    const subLayerProfilePresentFlag = [];
+    const subLayerLevelPresentFlag = [];
+    for (let i = 0; i < maxSubLayersMinus1; i++) {
+      subLayerProfilePresentFlag.push(gb.readBoolean());
+      subLayerLevelPresentFlag.push(gb.readBoolean());
+    }
+    if (maxSubLayersMinus1 > 0) {
+      for (let i = maxSubLayersMinus1; i < 8; i++) {
+        gb.readBits(2);
+      }
+    }
+    for (let i = 0; i < maxSubLayersMinus1; i++) {
+      if (subLayerProfilePresentFlag[i]) {
+        // sub_layer_profile_space,
```

Comment:
This section is hard to understand and inconsistently commented.  Can you clarify what fields these bytes are for?  Or use a skip function to skip N bytes if that's clearer?

Reply:
I have used a function to skip those bits :)